server/src/main/java/dev/slimevr/bridge/ProtobufBridge.java

```diff
@@ -125,6 +130,132 @@ protected void writeTrackerUpdate(Tracker localTracker) {
 		sendMessage(ProtobufMessage.newBuilder().setPosition(builder).build());
 	}
 
+	// Battery Status
+	@VRServerThread
+	protected void writeBatteryUpdate(Tracker localTracker) {
+		float lowestLevel = 200; // Arbitrarily higher than expected battery
+		// percentage
+		float trackerLevel = 0; // Tracker battery percentage on a scale from 0
+		// to 100. SteamVR expects a value from 0 to 1.
+		float trackerVoltage = 0; // Tracker voltage. This is used to determine
+		// if the tracker is being charged. owoTrack
+		// devices do not have a tracker voltage.
+		boolean isCharging = false;
+
+		List<Tracker> allTrackers = Main.getVrServer().getAllTrackers();
+		TrackerRole role = localTracker.getTrackerPosition().getTrackerRole();
+
+		Tracker primaryTracker = null;
+		Tracker secondaryTracker = null;
+		Tracker tertiaryTracker = null;
+
+		// Given what the role is of localTracker, the tracker positions that
+		// make up that role are set to primaryTracker, secondaryTracker, and
+		// tertiaryTracker respectively.
+		primaryTracker = TrackerUtils
+			.getNonInternalTrackerForBodyPosition(
+				allTrackers,
+				TrackerPosition.getByTrackerRole(role)
+			);
+		switch (role) {
+			case WAIST:
+				secondaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(allTrackers, TrackerPosition.WAIST);
+				break;
```

Comment:
Why is this getting the waist tracker again? Wouldn't both `primaryTracker` and `secondaryTracker` be the same tracker in this case?
It seems like you'd be wanting to do this instead:
```suggestion
			case WAIST:
				secondaryTracker = TrackerUtils
					.getNonInternalTrackerForBodyPosition(allTrackers, TrackerPosition.CHEST);
				tertiaryTracker = TrackerUtils
					.getNonInternalTrackerForBodyPosition(allTrackers, TrackerPosition.HIP);
				break;
```

Reply:
Chest should be its own tracking point in this case, as no other SlimeVR tracking points merge to a single SteamVR tracker for it. It should always be caught by:
`primaryTracker = TrackerUtils.getNonInternalTrackerForBodyPosition(allTrackers, TrackerPosition.getByTrackerRole(role))`.
Likewise, because there is no case for chest for `secondaryTracker`, it will never be part of multiple trackers for checking battery life.

> Wouldn't both `primaryTracker `and `secondaryTracker` be the same tracker in this case?

I think this is a result of confusion between `TrackerRole` and `BodyPart`. Hip is  `TrackerRole.WAIST`, and `BodyPart.HIP` whereas waist is null and `BodyPart.WAIST`. As I understand it, `TrackerPosition.getByTrackerRole(role)` should only return a tracker which is of `TrackerRole`, this should naturally exclude: waist, lower leg, and lower arm from being `primaryTracker` (because their `TrackerRole` is null), then leaving these aforementioned body parts to only be set as `secondaryTracker`. It is a bit confusing, I admit...

Your intuition is sort of correct though. This _could_ cause an issue with shoulder trackers in particular because they do not have a `TrackerRole` of null, however, uniquely this tracker position does not get a unique SteamVR tracking point, and is instead merged with elbow, so -- I believe -- there shouldn't be any issues with shoulder trackers overwriting the battery level of lower/upper arm trackers. I could be wrong on this, but as it is, shoulder + upper/lower arm is a very rare configuration and I haven't had anyone mention it as an issue; I should test this to see if it causes any issues.

Reply:
Chest should be its own tracking point in this case, as no other SlimeVR tracking points merge to a single SteamVR tracker for it. It should always be caught by:
`primaryTracker = TrackerUtils.getNonInternalTrackerForBodyPosition(allTrackers, TrackerPosition.getByTrackerRole(role))`.
Likewise, because there is no case for chest for `secondaryTracker`, it will never be part of multiple trackers for checking battery life.